From 745bace12e61d96295b6aa357940c72cc5152bcc Mon Sep 17 00:00:00 2001
From: Yang Chang <yang.chang@intel.com>
Date: Fri, 19 Jul 2013 11:10:29 +0800
Subject: Dalvik: immuneLimit should be NULL if RegionGC is disabled

BZ: 124332

immuneLimit should be NULL in stop-the-world GC if RegionGC optimization
is disabled. The current implementation doesn't consider the fact that
RegionGC optimization may be turned off dynamically by using -XnoRegionGC.

Category: device-enablement
Domain: AOSP-Dalvik-GC
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I14922c0edfb83b8c2f2ddb8d3b59b43f376750c1
Orig-MCG-Change-Id: I72429bd7e2695d2910d50d76c1115d098f791139
Signed-off-by: Yang Chang <yang.chang@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Reviewed-on: http://android.intel.com:8080/123674
Reviewed-by: Popov, Ivan G <ivan.g.popov@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/alloc/Heap.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/alloc/Heap.cpp b/vm/alloc/Heap.cpp
index 5ec7650..01020ad 100644
--- a/vm/alloc/Heap.cpp
+++ b/vm/alloc/Heap.cpp
@@ -557,7 +557,7 @@ void dvmCollectGarbageInternal(const GcSpec* spec)
     /* Set up the marking context.
      */
 #ifdef WITH_REGION_GC
-    if (!dvmHeapBeginMarkStep(spec->isPartial || spec->isRegional)) {
+    if (dvmHeapBeginMarkStep(spec->isPartial == true || (spec->isRegional == true && gDvm.enableRegionGC == true)) == false) {
 #else
     if (!dvmHeapBeginMarkStep(spec->isPartial)) {
 #endif
-- 
1.7.4.1

